src/client/dfs/dfs.c

```diff
@@ -1063,6 +1063,9 @@ entry_stat(dfs_t *dfs, daos_handle_t th, daos_handle_t oh, const char *name, siz
 		} else {
 			daos_handle_t	file_oh;
 
+			/* Added to avoid issues in daos_hhash_link_lookup due to uninitialized
+			 * file_oh in NLT fault injection test */
+			memset(&file_oh, 0, sizeof(daos_handle_t));
```

Comment:
There are several issues here:

1. Testing is to find bugs, when you fix one you're not "making the test pass", you're fixing a bug so you should reference the bug, not the test in any comment.
2. This comment includes technical information on a different part of the code-base, that might be appropriate for a commit message but it doesn't help future readers of this function understand the code in front of them.
3. Many daos functions, `daos_array_open_with_attr` included return a rc value and pass out the result via a arg, functions like this should always set the arg `oh` on the successful case and not use it on the unsuccessful case.  They should not expect the arg to be zeroed before being called.
4. If you did want to set file_oh to zeros then the correct way would be to change the declaration to be `daos_handle_t file_oh = {0}` or `daos_handle_t file_oh = {}`, I think the compiler would expect both.
5. Further to the 3. above `daos_array_open_with_attr` defines `of` as a param of type `out` so it's value at input should not matter.

What I think is happening here is that there's likely a bug in the error handling of array_open() which I think is exactly the type of bug this test is looking for.

Can you look at this again please and revert this change.

Reply:
Thank you! Initialize oh.cookie with an invalid handle in dc_obj_open() now.